packages/react/src/components/Authenticator/SetupTOTP/SetupTOTP.tsx

```diff
@@ -41,9 +41,8 @@ export const SetupTOTP = ({
   const [qrCode, setQrCode] = React.useState<string>();
   const [copyTextLabel, setCopyTextLabel] = React.useState<string>('COPY');
   const [secretKey, setSecretKey] = React.useState<string>('');
-
   const { totpIssuer = 'AWSCognito', totpUsername = user?.username } =
-    (fields as LegacyQRFields)?.QR ?? {};
```

Comment:
This isn't in the fields, since it gets deleted in the `formFields.ts` file. Otherwise it creates an unusual `QR` input field on the `SetupTOTP` page. Since the behavior of `QR` doesn't match up with our other override field logic, I created a new `QRFields` from the `useAuthenticator` hook, that retrieves this information to use.

Reply:
It's worth noting Angular and Vue do not use this logic, and garb the field data out of the xState context directly. React Native doesn't seem to allow issuer overrides, so this should not affect it.

Reply:
Curious, why doesn't current code work and whether it's a regression. If so, we should avoid adding an extra return value from `useAuthenticator`, and resolve that regression instead. 

Ideally, `fields` should be the source of truth. Is `fields` from `useAuthenticator` not getting updated properly?

Reply:
In the original implementation Vue, React and Angular all looked up the issuer information directly from the actor context in the SetupTOTP file. 

Even though the `QR` object is passed into the `formFields` prop it really has never been used as as form field, since it really isn't one. In fact it's deleted out of the form fields as seen below.

```
export const getFormFields = (
  route: FormFieldComponents,
  state: AuthMachineState
): FormFields => {
...
  delete formFields['QR'];
  return formFields;
```
When the big `react-core` refactor was done, it was assumed it was in the fields, but this was never the case. Vue and React still pulled it directly from the context.

If we keep it in the fields, then we'll need to update the `FormFields` components for each framework and ignore the `QR` field, since it doesn't exist. The `formFields` prop is used to help generate all form fields, even those that don't exist. 

I'm up for suggestions on a better way of handling this. I noticed in the refactor a `getTotpSecretCode` was added, so I assumed it was ok to add another TOTP/QR return value.

Reply:
@calebpollman Well QR isn't a formField, it's an image, but we kept all the logic for updating and customizing components, like dialCode, etc, in `formField`. In our next breaking change we can remove it from there.

Reply:
Return value from `useAuthenticator` has been removed.

Reply:
Reverted back, after offline discussion with @calebpollman - Since it affects React native. `useAuthenticator` is now updated with new `QRFields`